### PR TITLE
feat: add integrator id to deposit events

### PR DESCRIPTION
**Pull request by melisaguevara**
Add `integratorId` field to `v3FundsDeposited` events and `/deposits` endpoint query params.

### CODE REVIEW

File: packages/indexer/src/services/spokePoolIndexer.ts [hunk 1]
```diff
@@ -298,4 +308,21 @@ export class Indexer extends BaseIndexer {
       tokensBridgedEvents,
     };
   }
+
+  private async getIntegratorId(deposit: across.interfaces.DepositWithBlock) {
+    const INTEGRATOR_DELIMITER = "1dc0de";
```
Comment (daywiss) [inline packages/indexer/src/services/spokePoolIndexer.ts:313]: what does this come from?

File: packages/indexer-api/src/dtos/deposits.dto.ts [hunk 2]
```diff
@@ -7,14 +8,16 @@ const stringToInt = s.coerce(s.number(), s.string(), (value) =>
 export const DepositsParams = s.object({
   depositor: s.optional(s.string()),
   recipient: s.optional(s.string()),
+  originChainId: s.optional(stringToInt),
+  destinationChainId: s.optional(stringToInt),
   inputToken: s.optional(s.string()),
   outputToken: s.optional(s.string()),
-  integrator: s.optional(s.string()),
-  status: s.optional(s.string()),
+  integratorId: s.optional(s.string()),
```
Comment (daywiss) [inline packages/indexer-api/src/dtos/deposits.dto.ts:15]: what is an integrator?

File: packages/indexer/src/services/spokePoolIndexer.ts [hunk 3]
```diff
@@ -298,4 +308,21 @@ export class Indexer extends BaseIndexer {
...
+    const INTEGRATOR_ID_LENGTH = 4; // Integrator ids are 4 characters long
+    let integratorId = undefined;
+    const txn = await this.spokePoolProvider.getTransaction(
+      deposit.transactionHash,
+    );
+    const txnData = txn.data;
+    if (txnData.includes(INTEGRATOR_DELIMITER)) {
```
Comment (daywiss) [inline packages/indexer/src/services/spokePoolIndexer.ts:320]: ok so this is some kind of referral program or something

**Review by daywiss — APPROVED** [review]

File: packages/indexer-api/src/dtos/deposits.dto.ts (hunk 2 above)
Comment (amateima) [reply packages/indexer-api/src/dtos/deposits.dto.ts:15]: @daywiss 
Across wants to have better insights about the flow of funds, so it was added an optional `integrator` flag that can be appended to the transaction call data and delimited by a hardcoded delimiter when calling SpokePool's `deposit()` function. [Here](https://docs.google.com/spreadsheets/d/1K4kyyYffjFn8LpTPu-p9ub0CL25iPcJ7c1cvvLJ1BI8/edit?gid=0#gid=0) we maintain a list of integrators IDs and the mapping to the platform it is appended by.
We store this in the DB as `/deposits` endpoint should be able to filter deposits by integrator

File: packages/indexer-api/src/dtos/deposits.dto.ts [hunk 4]
```diff
@@ -7,14 +8,16 @@ const stringToInt = s.coerce(s.number(), s.string(), (value) =>
 export const DepositsParams = s.object({
   depositor: s.optional(s.string()),
   recipient: s.optional(s.string()),
+  originChainId: s.optional(stringToInt),
+  destinationChainId: s.optional(stringToInt),
```
Comment (amateima) [inline packages/indexer-api/src/dtos/deposits.dto.ts:12]: why are these necessary when requesting lists of deposits?

**Review by amateima — APPROVED** [review]
